components/PairingItem.js

```diff
@@ -13,13 +15,54 @@ import {
 } from "@/_styles";
 import { flavorColors } from "@/utils";
 import ingredientsData from "@/assets/ingredients.json";
+import CommentPopup from "@/components/CommentPopup";
+import StarRating from "./RatingStar";
 
 const PairingItem = ({ pairing, toggleFavoritePairing, isFavorite }) => {
+  const [showCommentPopup, setShowCommentPopup] = useState(false);
+  const [comments, setComments] = useState([]);
+  const [editingComment, setEditingComment] = useState(null);
+  const [rating, setRating] = useState(pairing.rating || 0);
+  const [totalRatings, setTotalRatings] = useState(pairing.totalRatings || 0);
+
   const ingredients = pairing.ingredients.map((id) => {
     const ingredient = ingredientsData.find((ing) => ing._id === id);
     return ingredient;
   });
 
+  const handleCommentSubmit = (comment, commentId) => {
+    if (commentId) {
+      const updatedComments = comments.map((c) =>
+        c.id === commentId ? { ...c, text: comment } : c
+      );
+      setComments(updatedComments);
+    } else {
+      setComments([...comments, { id: nanoid(), text: comment }]);
```

Comment:
I recommend changing your code to this to have more explicit naming:

```suggestion
  const handleCommentSubmit = (commentText, commentId) => {
    if (commentId) {
      const updatedComments = comments.map((comment) =>
        comment.id === commentId ? { ...comment, text: commentText } : comment
      );
      setComments(updatedComments);
    } else {
      setComments([...comments, { id: nanoid(), text: commentText }]);
```